test/tooling/db/db.go

```diff
@@ -0,0 +1,77 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package db
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/repository/schema"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	postgresImage    = "postgres:16-alpine"
+	postgresDatabase = "cartesinode"
+	postgresUsername = "admin"
+	postgresPassword = "password"
+)
+
+func Setup(ctx context.Context) (string, error) {
+	endpoint, ok := os.LookupEnv("TESTS_POSTGRES_ENDPOINT")
+	if !ok {
+		container, err := SetupContainer(ctx)
+		if err != nil {
+			return "", err
+		}
+
+		endpoint, err = container.ConnectionString(ctx, "sslmode=disable")
+		if err != nil {
+			return "", err
+		}
+	}
+
+	err := SetupSchema(endpoint)
+	if err != nil {
+		return "", err
+	}
+
+	return endpoint, nil
+}
+
+func SetupContainer(ctx context.Context) (*postgres.PostgresContainer, error) {
+	log := "database system is ready to accept connections"
```

Comment:
It would be better to use our "deps" package here. As in the future we want to use Github Services instead of Testcontainers. "deps" will serve as an intermediate layer. You can start only the Database by disabling Devnet

Reply:
Currently, `deps` does too much. I wouldn't like to mix it with this code, that is very self contained.

In the future, we should strip `deps` for parts and create smaller packages (probably in this very */tooling* folder).